internal/operators/ocs/ocs_operator.go

```diff
@@ -92,6 +94,43 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(context.Context, *common.Cluster, *models.Host) (*models.ClusterHostRequirementsDetails, error) {
-	return &models.ClusterHostRequirementsDetails{}, nil
+func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	numOfHosts := len(cluster.Hosts)
+
+	if numOfHosts < 3 {
+		return nil, fmt.Errorf("number of hosts is less than 3")
+	} else if numOfHosts == 3 { // Compact Mode
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores:   CPUCompactMode,
+				RAMMib:     MemoryCompactMode,
+				DiskSizeGb: MinDiskSize,
+			}, nil
+		}
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	} else if host.Role == models.HostRoleAutoAssign { //If auto-assign we cannot proceed with minimal or standard mode
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	} else {
+		// Minimum mode is the minimum version of the standard mode. If the resources will be more, standard mode will be deployed automatically.
+		// In minimum and standard mode we don't care about masters so returning zero
+		if host.Role == models.HostRoleMaster {
+			return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
+		}
+
+		var inventory *models.Inventory
+		_ = json.Unmarshal([]byte(host.Inventory), &inventory)
+
+		if getValidDiskCount(inventory.Disks) > 1 {
+			return &models.ClusterHostRequirementsDetails{
```

Comment:
I think the memory and CPU requirements per each disk should be added, like they are during the validation: https://github.com/openshift/assisted-service/blob/659b5dab9f676d4bdc70675c20f0b2d1111929ce/internal/operators/ocs/validations.go#L127

Reply:
@jakub-dzon , there can be different number of disks on different hosts. How do we know, whether to return `min,max,avg disk requirements` per host ? Hence it was decided that disk requirements will be taken care by cluster validation for OCS. WDYT?

Reply:
These requirements are returned in a context of a host, so you should know exact number of disks  - like during the validation.

Reply:
yes, but there can be different number of disks on different hosts. for eg, if we have 1,2,3 distribution of disks for 3 hosts, the requirements for each host will be different. What disk requirements should we return in that case?

Reply:
And that is expected - the UI will show that certain hosts need more CPU/RAM because they have different configuration; the API returns list of discovered hosts with their requirements.

Similar situation is with CNV  - certain hosts will have GPUs or SR-IOV NICs, which increase memory requirements for those hosts.

Reply:
The requirements are expected to be calculated for the exact host that is passed in the method parameter.

Reply:
Ohh Okay. got it. So each host can have different set of requirements, right?

Reply:
@rewantsoni  ^^

Reply:
That's right - the requirements don't have to be the same for all hosts in the cluster.